relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,153 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use relay_dynamic_config::GlobalConfig;
+use relay_system::{Addr, AsyncResponse, FromMessage, Interface, Sender, Service};
+use tokio::sync::watch;
+
+use crate::actors::project_upstream::{GetProjectStates, GetProjectStatesResponse};
+use crate::actors::upstream::{SendQuery, UpstreamRelay, UpstreamRequestError};
+
+/// Service implementing the [`GlobalConfigInterface`] interface.
+///
+/// The service offers two alternatives to fetch the [`GlobalConfig`]:
+/// responding to a [`GlobalConfigInterface::Get`] message with the config for
+/// one-off requests, or subscribing to updates with
+/// [`GlobalConfigInterface::Subscribe`] to keep up-to-date.
+#[derive(Debug)]
+pub struct GlobalConfigService {
+    /// Sender of the [`watch`] channel for the subscribers of the service.
+    // NOTE(iker): Placing the sender behind an Arc is a workaround to be able
+    // to send updates through this channel from invoked tasks where the global
+    // config request is made from upstream, since a watch sender cannot be
+    // cloned. To keep the latest config in the channel, it's assumed responses
+    // from the upstream are resolved faster than the fetch interval of this
+    // service. An alternative is to create a channel internal to the service
+    // and handle the updates through them.
+    sender: Arc<watch::Sender<Arc<GlobalConfig>>>,
+    /// Upstream service to request global configs from.
+    upstream: Addr<UpstreamRelay>,
+    /// Number of seconds to wait before making another request.
+    fetch_interval: u64,
```

Comment:
Please make this an `std::time::Duration`. A field `fetch_interval` does not convey which unit is in. Notice how we use `Duration` in all our interfaces (except for raw config structs, since it's not serializable; `Config` has accessor methods for this and clear documentation on all fields).